Geometry_Engine/Query/Length.cs

```diff
@@ -118,6 +202,81 @@ private static double Length(this ICurve curve)
             return double.NaN;
         }
 
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        [Description("Calculates the square binomial cooefficients used in the infinite series of the ellipse length calculation.")]
+        private static List<double> SquareBinomialCoefficientsEllipse(int requiredCount)
+        {
+            //Only allow for one thread to modify the cache at once.
+            lock (m_cooeficientLock)
+            {
+                //Calcuation of coefficients only needs to be done once.
+                //Coefficients are being stored for next ellipse to be evaluated
+                if (m_ellipseCoeficients == null)
```

Comment:
Couldn't you declare the initial value of `m_ellipseCoeficients = new List<double>();`?

Reply:
true, will fix. Leftover from previous iteration of the code where the coefficient list was always fully populated.